src/main/java/co/rsk/bitcoinj/core/StoredBlock.java

```diff
@@ -113,12 +121,37 @@ public StoredBlock getPrev(BtcBlockStore store) throws BlockStoreException {
         return store.get(getHeader().getPrevBlockHash());
     }
 
-    /** Serializes the stored block to a custom packed format. Used by {@link CheckpointManager}. */
+    /**
+     * Serializes the stored block to a custom packed format. Used internally.
+     * As of June 22, 2024, it takes 12 unsigned bytes to store the chain work value,
+     * so developers should use the V2 format.
+     *
+     * @param buffer buffer to write to
+     */
     public void serializeCompact(ByteBuffer buffer) {
```

Comment:
Should this be renamed to `serializeCompactV1` or `serializeCompactLegacy` and add a @deprecated tag maybe?

It should be simple to know which method to use

Reply:
Updated